reana_client/utils.py

```diff
@@ -208,9 +208,34 @@ def _validate_environment(reana_yaml):
         workflow_steps = reana_yaml["workflow"]["steps"]["stages"]
         _validate_yadage_workflow_environment(workflow_steps)
     elif workflow_type == "cwl":
-        logging.warning(
-            "The environment validation is not implemented for CWL workflows yet."
+        workflow_file = reana_yaml["workflow"].get("file")
+        _validate_cwl_workflow_environment(workflow_file)
+
+
+def _validate_cwl_workflow_environment(workflow_file):
+    """Validate environments in REANA CWL workflow.
+
+    :param workflow_file: Path to CWL workflow specification.
+    :raises Warning: Warns user if the workflow environment is invalid in CWL workflow steps.
+    """
+    try:
+        import cwl_utils.parser_v1_0 as cwl_parser
+        from cwl_utils.docker_extract import traverse
+    except ImportError as e:
+        click.secho(
+            "==> ERROR: Environment image validation using Python 2.7 is not supported for CWL workflows.",
```

Comment:
Minor: I think it would not be clear what action to take if you see this message, I would rather specifically say what to do. Something along these lines:
`==> ERROR: Cannot validate environment. Please install reana-client on Python 3+ to enable environment validation for CWL workflows.`